### PR TITLE
Elasticsearch: add fail_on_detecting_es_version_retry_exceed support

### PR DESCRIPTION
**Pull request by hyperion-hyp**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?
Add fail_on_detecting_es_version_retry_exceed support for [elasticsearch output](https://github.com/uken/fluent-plugin-elasticsearch#fail_on_detecting_es_version_retry_exceed).

### Why?
This field is necessary to prevent Fluend from failing at startup when one or many Elasticsearch hosts are unreachable.

### Additional context
Related:
- https://github.com/banzaicloud/logging-operator/issues/491
- https://github.com/uken/fluent-plugin-elasticsearch/issues/475
- 

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

**Review by ahma — APPROVED** [review]
LGTM Thanks @hyperion-hyp

File: pkg/sdk/logging/model/output/elasticsearch.go [hunk 1]
```diff
@@ -142,6 +142,9 @@ type ElasticsearchOutput struct {
 	// Indicates whether to fail when max_retry_putting_template is exceeded. If you have multiple output plugin, you could use this property to do not fail on fluentd statup.(default: true)
 	// +kubebuilder:validation:Optional
 	FailOnPuttingTemplateRetryExceed *bool `json:"fail_on_putting_template_retry_exceed,omitempty" plugin:"default:true"`
+        // fail_on_detecting_es_version_retry_exceed (default: true)
+        // +kubebuilder:validation:Optional
+        FailOnDetectingEsVersionRetryExceed *bool `json:"fail_on_detecting_es_version_retry_exceed,omitempty" plugin:"default:true"`
```
Comment (siliconbrain) [inline pkg/sdk/logging/model/output/elasticsearch.go:147]: use tabs to indent
```suggestion
	// fail_on_detecting_es_version_retry_exceed (default: true)
	// +kubebuilder:validation:Optional
	FailOnDetectingEsVersionRetryExceed *bool `json:"fail_on_detecting_es_version_retry_exceed,omitempty" plugin:"default:true"`
```

Comment (hyperion-hyp) [reply pkg/sdk/logging/model/output/elasticsearch.go]: done
@siliconbrain thanks

**Review by siliconbrain — DISMISSED** [review]

**Comment by siliconbrain** [comment]
@hyperion-hyp to fix the failing build, you should run `make generate` and commit the changes

**Comment by hyperion-hyp** [comment]
> @hyperion-hyp to fix the failing build, you should run `make generate` and commit the changes
... still failing; what did I miss?

**Comment by siliconbrain** [comment]
@hyperion-hyp seems like you've moved the field from before `fail_on_detecting_es_version_retry_exceed` to after it sometime after you last ran generate and `controller-gen` now mirrors this change in the manifest files. Run
```
make manifests        # regenerates manifest files
make generate         # regenerates vfs
```
and commit any changes

**Comment by hyperion-hyp** [comment]
@siliconbrain, thanks for your guidance, i've made some progress, but still no luck.

**Comment by siliconbrain** [comment]
@hyperion-hyp you should check the [build details](https://app.circleci.com/pipelines/github/banzaicloud/logging-operator/3731/workflows/e4decdff-7298-4aff-9a61-babe6b5e27fe/jobs/10119) 😉 it says that your newly added field (which is present by default) is missing from a config generation test's expected output:
```
⋮
--- FAIL: TestAwsElasticsearch (0.00s)
⋮
         exception_backup true
        -fail_on_detecting_es_version_retry_exceed true
         fail_on_putting_template_retry_exceed true
⋮
```

**Comment by hyperion-hyp** [comment]
done!
tnx @siliconbrain

**Comment by siliconbrain** [comment]
@hyperion-hyp I hate to be the bearer of bad news...again 😅 but it seems that you have a merge conflict in the generated vfs code because another PR has been merged to master which also changed some custom resources. To resolve this, please rebase your branch onto master OR merge master into your branch, then run
```sh
make generate        # regenerate any DeepCopy implementations
make manifests       # regenerates manifest files
make generate         # regenerates vfs
```
and commit and push (or force push, in case of a rebase) your changes.

**Review by aslafy-z — APPROVED** [review]
LGTM

**Comment by hyperion-hyp** [comment]
@siliconbrain, A rainbow follows every bad news :), thanks a lot!

**Review by siliconbrain — APPROVED** [review]
👍

**Review by ahma — APPROVED** [review]

**Pull request merged**